libs/diag/sdk_error.go

```diff
@@ -0,0 +1,61 @@
+package diag
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+)
+
+func findApiErr(e error) *apierr.APIError {
+	for {
+		cast, ok := e.(*apierr.APIError)
+		if ok {
+			return cast
+		}
+
+		inner := errors.Unwrap(e)
+		if inner == nil {
+			break
+		}
+		e = inner
+	}
+	return nil
+}
+
+func FormatAPIErrorSummary(e error) string {
+	extra := ""
+	apiErr := findApiErr(e)
+	if apiErr != nil {
```

Comment:
I recommend inverting this: return early if it isn't an API error. Function below does the same.